### PR TITLE
Adds support for RKE2

### PR DESCRIPTION
**Pull request by briandowns**
Updates the pod spec to allow for hostIPC, hostPID, and necessary volume
mounts.

Reference issue: https://github.com/rancher/rke2/issues/1

### CODE REVIEW

**Comment by brandond** [comment]
LGTM once the comment from @prachidamle has been addressed

File: pkg/securityscan/job/job.go [hunk 1]
```diff
@@ -69,6 +69,8 @@ func New(clusterscan *cisoperatorapiv1.ClusterScan, clusterscanprofile *cisopera
 					},
 				},
 				Spec: corev1.PodSpec{
+					HostPID:                       true,
+					HostIPC:                       true,
```
[the remark this answers is not in the document]
Comment (briandowns) [reply pkg/securityscan/job/job.go:73]: We're good to go on scanning RKE. 
image

**Review by brandond — APPROVED** [review]

**Review by prachidamle — APPROVED** [review]
LGTM

File: pkg/securityscan/job/job.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (prachidamle) [reply pkg/securityscan/job/job.go:73]: Thank you for testing!!

**Review by dweomer — APPROVED** [review]

**Pull request merged**